omnigibson/utils/asset_conversion_utils.py

```diff
@@ -0,0 +1,2481 @@
+    for prim in obj_prim.GetChildren():
+        looks_prim = None
+        if prim.GetName() == "Looks":
+            looks_prim = prim
+        elif prim.GetPrimTypeInfo().GetTypeName() == "Xform":
+            looks_prim_path = f"{str(prim.GetPrimPath())}/Looks"
+            looks_prim = lazy.omni.isaac.core.utils.prims.get_prim_at_path(looks_prim_path)
+        if not looks_prim:
+            continue
+        for subprim in looks_prim.GetChildren():
+            if subprim.GetPrimTypeInfo().GetTypeName() != "Material":
+                continue
+            log.debug(
+                f"Removed material prim {subprim.GetPath()}:",
+                stage.RemovePrim(subprim.GetPath()),
+            )
+
+    # # Create new default material for this object.
+    # mtl_created_list = []
+    # lazy.omni.kit.commands.execute(
+    #     "CreateAndBindMdlMaterialFromLibrary",
+    #     mdl_name="OmniPBR.mdl",
+    #     mtl_name="OmniPBR",
+    #     mtl_created_list=mtl_created_list,
+    # )
+    # default_mat = lazy.omni.isaac.core.utils.prims.get_prim_at_path(mtl_created_list[0])
+    # default_mat = rename_prim(prim=default_mat, name=f"default_material")
+    # log.debug("Created default material:", default_mat.GetPath())
+    #
+    # # We may delete this default material if it's never used
+    # default_mat_is_used = False
+
+    # Grab all visual objs for this object
+    urdf_path = f"{dataset_root}/objects/{obj_category}/{obj_model}/{obj_model}_with_metalinks.urdf"
+    visual_objs = _get_visual_objs_from_urdf(urdf_path)
+
+    # Extract absolute paths to mtl files for each link
+    link_mtl_files = OrderedDict()  # maps link name to dictionary mapping mesh name to mtl file
+    mtl_infos = OrderedDict()  # maps mtl name to dictionary mapping material channel name to png file
+    mat_files = OrderedDict()  # maps mtl name to corresponding list of material filenames
+    mtl_old_dirs = OrderedDict()  # maps mtl name to corresponding directory where the mtl file exists
+    mat_old_paths = OrderedDict()  # maps mtl name to corresponding list of relative mat paths from mtl directory
+    for link_name, link_meshes in visual_objs.items():
+        link_mtl_files[link_name] = OrderedDict()
+        for mesh_name, obj_file in link_meshes.items():
+            # Get absolute path and open the obj file if it exists:
+            if obj_file is not None:
+                obj_path = f"{dataset_root}/objects/{obj_category}/{obj_model}/{obj_file}"
+                with open(obj_path, "r") as f:
+                    mtls = []
+                    for line in f.readlines():
+                        if "mtllib" in line and line[0] != "#":
+                            mtls.append(line.split("mtllib ")[-1].split("\n")[0])
+
+                if mtls:
+                    assert len(mtls) == 1, f"Only one mtl is supported per obj file in omniverse -- found {len(mtls)}!"
+                    mtl = mtls[0]
+                    # TODO: Make name unique
```

Comment:
is this TODO critical?

Reply:
Oh this is a super old todo from years ago, and this function currently isn't used at all (we don't import rendering channels)